src/context.ts

```diff
@@ -1,35 +1,35 @@
 import { Addresses } from "./entities";
 
 export class EdcConnectorClientContext implements Addresses {
-  #apiToken: string;
+  #apiToken: string | undefined;
   #addresses: Addresses;
 
-  constructor(apiToken: string, addresses: Addresses) {
+  constructor(apiToken: string | undefined, addresses: Addresses) {
     this.#apiToken = apiToken;
     this.#addresses = addresses;
   }
 
   get default(): string {
-    return this.#addresses.default;
+    return this.#addresses.default!;
```

Comment:
With this change, we are losing some type-safety (already largely missing in JS). If the field is optional, then we should treat it as optional as done for the `apiToken`

Reply:
yes, this was a quick fix to have the compilation working, this should return a `string | undefined` type